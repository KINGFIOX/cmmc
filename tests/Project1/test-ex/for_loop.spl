int main() {
    int i;
    for(i=0;i<10;i=i+1)
        write(i);
    i = 0;
    for(;;) {
        i = i + 1;
        if(i == 5)
            continue;
        if(i >= 10)
            break;
    }
    return i;
}
