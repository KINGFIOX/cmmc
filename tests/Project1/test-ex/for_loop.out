Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: i
            SEMI
        StmtList (3)
          Stmt (3)
            FOR
            LP
            Stmt (3)
              Exp (3)
                Exp (3)
                  ID: i
                ASSIGN
                Exp (3)
                  INT: 0
              SEMI
            ForOptional (3)
              Exp (3)
                Exp (3)
                  ID: i
                LT
                Exp (3)
                  INT: 10
            SEMI
            ForOptional (3)
              Exp (3)
                Exp (3)
                  ID: i
                ASSIGN
                Exp (3)
                  Exp (3)
                    ID: i
                  PLUS
                  Exp (3)
                    INT: 1
            RP
            Stmt (4)
              Exp (4)
                Exp (4)
                  ID: write
                LP
                Args (4)
                  Exp (4)
                    ID: i
                RP
              SEMI
          StmtList (5)
            Stmt (5)
              Exp (5)
                Exp (5)
                  ID: i
                ASSIGN
                Exp (5)
                  INT: 0
              SEMI
            StmtList (6)
              Stmt (6)
                FOR
                LP
                Stmt (6)
                  SEMI
                SEMI
                RP
                Stmt (6)
                  CompSt (6)
                    LC
                    StmtList (7)
                      Stmt (7)
                        Exp (7)
                          Exp (7)
                            ID: i
                          ASSIGN
                          Exp (7)
                            Exp (7)
                              ID: i
                            PLUS
                            Exp (7)
                              INT: 1
                        SEMI
                      StmtList (8)
                        Stmt (8)
                          IF
                          LP
                          Exp (8)
                            Exp (8)
                              ID: i
                            EQ
                            Exp (8)
                              INT: 5
                          RP
                          Stmt (9)
                            CONTINUE
                            SEMI
                        StmtList (10)
                          Stmt (10)
                            IF
                            LP
                            Exp (10)
                              Exp (10)
                                ID: i
                              GE
                              Exp (10)
                                INT: 10
                            RP
                            Stmt (11)
                              BREAK
                              SEMI
                    RC
              StmtList (13)
                Stmt (13)
                  RETURN
                  Exp (13)
                    ID: i
                  SEMI
        RC
